src/AppServices/DataSciencePage.tsx

```diff
@@ -1,55 +1,410 @@
+        </TextContent>
+      </StackItem>
+    </Stack>
+  );
+};
+
+const UpgradeClusterModalContent = ({
+  handleStrategyChange,
+  upgradeStrategy,
+  ...props
+}: ClusterSelectProps & {
+  upgradeStrategy: UpgradeStrategy;
+  handleStrategyChange: (strategy: UpgradeStrategy) => void;
+}) => {
+  const { t } = useTranslation("datascienceoverview");
+  const upgradeBody = (
+    <TextContent>
+      <Text component="small">{t("upgradeClusterSelectDescription")}</Text>
+      <ClusterSelect
+        {...props}
+        isDisabled={upgradeStrategy !== UpgradeStrategy.Upgrade}
+      />
+      <TextContent>
+        <Text>TODO: Add requrements values</Text>
+      </TextContent>
+    </TextContent>
+  );
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <Radio
+          id="upgrade-cluster"
+          name="upgrade-cluster"
+          label={t("upgradeClusterSelectLabel")}
+          onChange={() => handleStrategyChange(UpgradeStrategy.Upgrade)}
+          isChecked={upgradeStrategy === UpgradeStrategy.Upgrade}
+          body={upgradeBody}
+        />
+      </StackItem>
+      <StackItem>
+        <Radio
+          id="create-cluster"
+          name="create-cluster"
+          label={t("createClusterLabel")}
+          onChange={() => handleStrategyChange(UpgradeStrategy.Create)}
+          isChecked={upgradeStrategy === UpgradeStrategy.Create}
+        />
+      </StackItem>
+    </Stack>
+  );
+};
+
+const CreateClusterContent = () => {
+  const { t } = useTranslation("datascienceoverview");
+  return (
+    <TextContent>
+      <Text>{t("createClusterContent")}</Text>
+    </TextContent>
+  );
+};
+
+const ClusterModalContent = ({
+  mode,
+  clusters,
+  handleSelectCluster,
+  selectedCluster,
+  upgradeStrategy,
+  handleStrategyChange,
+}: {
+  selectedCluster?: ClusterObject;
+  handleSelectCluster: HandleSelectCluster;
+  mode: RHodsClusterAddonMode;
+  clusters: ClusterObject[];
+  upgradeStrategy: UpgradeStrategy;
+  handleStrategyChange: (strategy: UpgradeStrategy) => void;
+}) => {
+  if (mode === RHodsClusterAddonMode.Detecting) {
+    return (
+      <Bullseye>
+        <Spinner size="xl" />
+      </Bullseye>
+    );
+  }
+
+  if (mode === RHodsClusterAddonMode.Install) {
+    return (
+      <InstallClusterModalContent
+        selectedCluster={selectedCluster}
+        handleSelectCluster={handleSelectCluster}
+        clusters={clusters}
+      />
+    );
+  }
+
+  if (mode === RHodsClusterAddonMode.Upgrade) {
+    return (
+      <UpgradeClusterModalContent
+        selectedCluster={selectedCluster}
+        handleSelectCluster={handleSelectCluster}
+        clusters={clusters}
+        upgradeStrategy={upgradeStrategy}
+        handleStrategyChange={handleStrategyChange}
+      />
+    );
+  }
+
+  if (mode === RHodsClusterAddonMode.Create) {
+    return <CreateClusterContent />;
+  }
+
+  return null;
+};
+
+export const DataSciencePage = ({ loadClusters }: DataSciencePageProps) => {
```

Comment:
```suggestion
export const DataSciencePage: VoidFunctionComponent<DataSciencePageProps> = ({ loadClusters }) => {
```

Reply:
I picked this one as it was quick to write a suggestion for, but I'd like all components to be explicitly typed as such.